### PR TITLE
NFTokenAcceptOffer wrong buyer in broker mode

### PR DESCRIPTION
**Pull request by shawnxie999**
### Context of Change
In brokered mode, the buyer inside simple view for `NFTokenAcceptOffer` is incorrect due to 2 `NFTokenOffer` nodes
- We can get the seller and buy by checking if the `Flags` field is 1 or 0 respectively

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### TypeScript/Hooks Update

- [ ] Updated files to React Hooks
- [ ] Updated files to TypeScript

## Before / After
Before
image

After
image

### CODE REVIEW

**Comment by shawnxie999** [comment]
I think another way of evaluating buy/sell off is to check if `Flags` is 1 or 0, instead of the `Destination` field. Not sure which way is more robust

**Comment by mvadari** [comment]
Flags are probably more robust

File: src/containers/shared/components/Transaction/NFTokenAcceptOffer/parser.ts [hunk 1]
```diff
@@ -18,6 +22,29 @@ export const parser: TransactionParser<
     acceptedOfferIDs.push(tx.NFTokenSellOffer)
   }
 
+  if (acceptedOfferNodes.length === 1) {
+    acceptedOfferNode = acceptedOfferNodes[0].DeletedNode?.FinalFields
+  } else if (acceptedOfferNodes.length > 1) {
```
[the remark this answers is not in the document]
Comment (shawnxie999) [reply src/containers/shared/components/Transaction/NFTokenAcceptOffer/parser.ts:27]: Hmm do you mean something like...

declaring
`
acceptedOfferNodes: { sellOfferNode: any; buyOfferNode: any }
`

[the remark this answers is not in the document]
Comment (mvadari) [reply src/containers/shared/components/Transaction/NFTokenAcceptOffer/parser.ts:27]: I was thinking something like 
```
const buyOfferNode = acceptedOfferNode
const sellOfferNode = acceptedOfferNode
```
Then flip the logic for when there are multiple nodes.

[the remark this answers is not in the document]
Comment (shawnxie999) [reply src/containers/shared/components/Transaction/NFTokenAcceptOffer/parser.ts:27]: sry could you elaborate? how would that benefit us?

[the remark this answers is not in the document]
Comment (mvadari) [reply src/containers/shared/components/Transaction/NFTokenAcceptOffer/parser.ts:27]: I just think the code would be neater and a little more DRY. Something like this:
```js
if (acceptedOfferNodes.length === 1) {
  const buyOfferNode = acceptedOfferNodes[0].DeletedNode?.FinalFields
  const sellOfferNode = acceptedOfferNodes[0].DeletedNode?.FinalFields
} else {
  // the same logic that you currently have for distinguishing between them
}
return {
    amount: formatAmount(buyOfferNode.Amount),
    tokenID: buyOfferNode.NFTokenID,
    seller: sellOfferNode.Owner,
    buyer: buyOfferNode.Owner,
    acceptedOfferIDs,
  }

[the remark this answers is not in the document]
Comment (mvadari) [reply src/containers/shared/components/Transaction/NFTokenAcceptOffer/parser.ts:27]: Hm, maybe that wouldn't work - the seller and buyer would be the same. But perhaps there's a way to merge the logic of the two situations a bit more so it's not repeated as much.

[the remark this answers is not in the document]
Comment (shawnxie999) [reply src/containers/shared/components/Transaction/NFTokenAcceptOffer/parser.ts]: I've cleaned up the code a bit to reduce if-statements. I think it's almost as best as possible without having to do a big refactor.

**Review by mvadari — APPROVED** [review]

File: src/containers/shared/components/Transaction/NFTokenAcceptOffer/parser.ts [hunk 2]
```diff
@@ -18,6 +20,29 @@ export const parser: TransactionParser<
...
+    return {
+      amount: formatAmount(buyOfferNode.Amount),
+      tokenID: buyOfferNode.NFTokenID,
+      seller: sellOfferNode.Owner,
+      buyer: buyOfferNode.Owner,
+      acceptedOfferIDs,
+    }
+  }
+
+  const acceptedOfferNode = acceptedOfferNodes[0]?.DeletedNode?.FinalFields
```
Comment (ckniffen) [inline src/containers/shared/components/Transaction/NFTokenAcceptOffer/parser.ts:44]: What will be showin if the accept fails and there is no offer in the meta data?  Will it show the wrong value for the buyer?

Comment (shawnxie999) [reply src/containers/shared/components/Transaction/NFTokenAcceptOffer/parser.ts:44]: The logic would be identical as before. The conditional check right after will be responsible for handling undefined if node doesnt exist.
`  if (!acceptedOfferNode) {
    return {
      acceptedOfferIDs,
    }
  }`

**Review by ckniffen — APPROVED** [review]

**Pull request merged**